src/warc2zim/url_rewriting.py

```diff
@@ -276,32 +271,58 @@ def __call__(
         The url is "fully" rewrited to point to a normalized entry path
         """
 
-        item_url = item_url.strip()
+        try:
+            item_url = item_url.strip()
 
-        item_scheme = urlsplit(item_url).scheme
-        if item_scheme and item_scheme not in ("http", "https"):
-            return item_url
+            item_scheme = urlsplit(item_url).scheme
+            if item_scheme and item_scheme not in ("http", "https"):
+                return item_url
 
-        item_absolute_url = urljoin(
-            urljoin(self.article_url.value, base_href), item_url
-        )
-        item_fragment = urlsplit(item_absolute_url).fragment
-
-        item_path = normalize(HttpUrl(item_absolute_url))
-
-        if rewrite_all_url or item_path in self.existing_zim_paths:
-            return self.get_document_uri(item_path, item_fragment)
-        else:
-            if (
-                self.missing_zim_paths is not None
-                and item_path not in self.missing_zim_paths
-            ):
-                logger.debug(f"WARNING {item_path} ({item_url}) not in archive.")
-                # maintain a collection of missing Zim Path to not fill the logs with
-                # duplicate messages
-                self.missing_zim_paths.add(item_path)
-            # The url doesn't point to a known entry
-            return item_absolute_url
+            item_absolute_url = urljoin(
+                urljoin(self.article_url.value, base_href), item_url
+            )
+
+            item_fragment = urlsplit(item_absolute_url).fragment
+
+            item_path = normalize(HttpUrl(item_absolute_url))
+
+            if rewrite_all_url or item_path in self.existing_zim_paths:
+                return self.get_document_uri(item_path, item_fragment)
+            else:
+                if (
+                    self.missing_zim_paths is not None
+                    and item_path not in self.missing_zim_paths
+                ):
+                    logger.debug(f"WARNING {item_path} ({item_url}) not in archive.")
+                    # maintain a collection of missing Zim Path to not fill the logs
+                    # with duplicate messages
+                    self.missing_zim_paths.add(item_path)
+                # The url doesn't point to a known entry
+                return item_absolute_url
+
+        except Exception as exc:
+            if logger.isEnabledFor(logging.DEBUG):
```

Comment:
There's no heavy processing here so running this conditionally isn't justified

Reply:
Agreed !